### DISCUSSION TITLE
What about 'testmanagerd' service on iOS17

### DISCUSSION THREAD

**Discussion by Rodionp13** [doronz88/pymobiledevice3#550] [Q&A] [answered]
Hi @doronz88, Could you please answer several questions regarding these 2 services ?
```
"com.apple.dt.testmanagerd.remote": {
      "Entitlement": "com.apple.private.dt.testmanagerd.client",
      "Port": "55299",
      "Properties": {
        "UsesRemoteXPC": false
      }
    },
    "com.apple.dt.testmanagerd.remote.automation": {
      "Entitlement": "AppleInternal",
      "Port": "55260",
      "Properties": {
        "UsesRemoteXPC": false
      }
    }
```
- Their "UsesRemoteXPC" key is false meaning they are not accessible via remoteXPC ? 
- And what about "Entitlement" key : It's not a `com.apple.mobile.lockdown.remote.trusted` meaning they won't be spawned via `lockdownd` on demand of `remoted` ? 
- So the main question is how can I connect to `testmanagerd` and setup test session ?

Sorry, if I'm wrong in my assumptions and thanks in advance !

**Comment by doronz88** [comment, marked as answer]
I think this should provide all answers:
https://github.com/doronz88/pymobiledevice3/blob/master/misc/RemoteXPC.md#accessing-services-over-the-trusted-tunnel

**Reply by Rodionp13** [reply]
thanks, I've read this again. And just to be sure: in case of `testmanagerd` I need to send this request, right ?
```
{
  "Label": "userAgent",
  "ProtocolVersion": "2",
  "Request": "RSDCheckin",
  "EscrowBag": "if any..."
}
```

**Reply by doronz88** [reply]
Correct. The rest is just normal legacy protocols
